_posts/2019-09-25-scatter-gather-pt1.md

```diff
@@ -0,0 +1,94 @@
+
+Now, there are a few differences between the available saga persisters which boils down to whether they use pessimistic or optimistic concurrency control to resolve concurrency issues on the same saga instance.** 
+* With optimistic concurrency control, concurrent access to the same saga data is only noticed once the message handler has completed and tries to update the saga's state. At this point, a concurrency exception will be raised by the database, failing the message. Recoverability mechanism will then ensure that the message is retried. You will notice some sort of concurrency related exceptions in your endpoint's log files if that occurs.
+* With pessimistic concurrency control, concurrent attempts to load the same sagas are blocked (by using some form of locks) to ensure sequential access to the data. This means that there will most likely be no exceptions visible. However, the endpoint message processing throughput will slow down significantly as in the worst case only one message can be processed at the same time. This means that with pessimistic concurrency control, the endpoint can fall down to a concurrency level of one, allowing only a single message to complete after another. This can also cause database servers to assume a deadlock if the locks are held for too long, in which case you'd also see exceptions showing up.
+
+While neither of the solutions is particularly helpful in this case, optimistic concurrency can make the situation a lot worse. Assuming that many messages to the same saga are in the queue, retrying a message due to a concurrency exception won't help much, as it will just enter a new race condition every time. After failing multiple times, recoverability will fail the message and move it to the error queue. This means the user has to manually retry the message later on instead of recoverability handling the issue automatically.
+
+I hope you now have a better understanding of why a Saga is a bad choice to implement the Scatter-Gather patten in almost all cases. The only exceptions might be situations where the responses trickle in slowly after each other than in parallel or when you're only scattering a very low amount of messages (< 5). Sagas are made to implement long-running processes and while their API might invite for other applications sometimes, there is a high chance that it ends up being a bad choice.
+
+We will look into a better solution for the Scatter-Gather pattern in the next blog post.
+
+
+*There is no absolute guarantee that the outgoing messages are atomic with completing the incoming message unless you're using a transaction mode of `SendsAtomicWithReceive` or higher. There is always some edge case that might interrupt the actual dispatch operation (e.g. a network failure in the middle of dispatching the batch), but NServiceBus does it's best to keep the window for such edge cases as small as possible. This is a topic for another blog post though.
```

Comment:
I wanted to avoid this level of technical details on a post about sagas but felt that the linked statement might give some wrong impressions. I think removing it might makes sense